### ISSUE TITLE
[0.12.0] distinct() and count() functions do not work on string fields

### ISSUE THREAD

**Issue by cnelissen** [influxdata/kapacitor#495]
I have a TickScript like:

```
stream
    |from()
        .database('mydb')
        .measurement('a')
    |distinct('stringField')
    |count('stringField')
```

and I am seeing errors in the logs like:

```
[edge:test|stream1->distinct2] 2016/04/28 20:08:08 I! aborting c: 22 e: 1
[test:distinct2] 2016/04/28 20:08:08 E! invalid field type: string
[edge:test|stream->srcstream0] 2016/04/28 20:08:08 D! closing c: 16208 e: 16208
[edge:test|srcstream0->stream1] 2016/04/28 20:08:08 D! closing c: 16208 e: 16208
[edge:test|stream1->distinct2] 2016/04/28 20:08:08 D! closing c: 22 e: 1
[task_master] 2016/04/28 20:08:08 E! Stopped task: test invalid field type: string
[task_store] 2016/04/28 20:08:08 E! task test finished with error: invalid field type: string
```

I posted a message on the mailing list and I got a reply from @nathanielc that said:

"Kapacitor does not support string fields yet. It won't be hard to add, it just hasn't been done yet."

This is a showstopper for me since my queries depend on getting an accurate count based on a string field. I didn't see an open (or closed) issue about this but it seems like it is a known issue...

Any ETA on a fix for this?

**Comment by nathanielc** [comment]
@cnelissen Thanks for filing an issue. The 0.13 milestone is coming to a close but I'll make this a priority for 0.14.

**Comment by cnelissen** [comment]
What is the timeframe for v0.14? Do you have any ideas for a workaround? I am pretty much dead in the water without being able to query string fields...

**Comment by nathanielc** [comment]
@cnelissen For a work around you could write a UDF to map strings to an int64s. You could use a hashing mechanism of if you know the static set via a direct mapping. Then Kapacitor will work well with the int64s. Then you could use another UDF to reverse the mapping back to strings.

See https://github.com/influxdata/kapacitor/tree/master/udf/agent/examples for example UDFs in Go and Python.

**Comment by cnelissen** [comment]
@nathanielc Thanks for the ideas! I think I might have a solution in place.

**Comment by nathanielc** [comment]
This was fixed in #524
